src/components/Main.jsx

```diff
@@ -0,0 +1,160 @@
+import axios from "axios";
+import { Box, Grid, GridItem, useDisclosure } from "@chakra-ui/react";
+import Filters from "./filter/Filters";
+import ProductList from "./pages/ProductList";
+import SortBy from "./filter/children/SortBy";
+import FilterListByType from "./filter/children/FilterListByType";
+import FilterTemplate from "./templates/FilterTemplate";
+
+const Main = () => {
+  const [productList, setProductList] = useState([]);
+  const [sortType, setSortType] = useState("asc");
+  const [listFilters, setListFilters] = useState([]);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [loading, setLoading] = useState(true);
+  const btnRef = useRef();
+  const [sortField, setSortField] = useState(400);
+  const [priceRange, setPriceRange] = useState([0, 1000]);
+  const tabs = [
+    { label: "Low to High", value: "asc", type: "sort" },
+    { label: "High to Low", value: "desc", type: "sort" },
+    { label: "Newest first", value: "createdAt", type: "sort" },
+  ];
+  // const API_ENDPOIT = import.meta.env.VITE_PUBLIC_API_URL;
+  useEffect(() => {
+    const fetchData = async () => {
+      setLoading(true);
+      const filteredBrands = FilterListByType(listFilters, "brand");
+      const filteredRatings = FilterListByType(listFilters, "rating");
+
+      try {
+        const response = await axios.post(
+          `${import.meta.env.VITE_PUBLIC_API_URL}/api/printers/filter`,
+          {
+            brand: filteredBrands,
+            rating: filteredRatings,
+            sortOrder: sortType,
+            sortField: sortField,
+            minPrice: priceRange[0],
+            maxPrice: priceRange[1],
+          }
+        );
+        console.log(response.data);
+        setProductList(response.data.data);
+      } catch (err) {
+        console.error(err);
+      } finally {
+        setLoading(false);
+      }
+    };
+    fetchData();
+  }, [listFilters, sortType, sortField, priceRange]);
+
+  const handleItem = (item) => {
+    setListFilters((prev) => {
+      if (!prev.some((i) => i.value === item.value)) {
+        return [...prev, item];
+      }
+      return prev.filter((ele) => ele.value !== item.value);
+    });
+  };
+  // remove a particular filter
+  const handleCloseFilter = (item) => {
+    setListFilters((prev) => prev.filter((i) => i.value !== item.value));
+  };
+
+  const handleClearAllFilters = () => {
+    setListFilters([]);
+  };
+
+  const handleSortChange = (sortValue) => {
+    setSortType(sortValue);
+    setSortField("price");
+  };
+
+  const handlePriceChange = (priceRange) => {
+    setPriceRange(priceRange);
+  };
+
+  return (
+    <Grid
+      templateAreas={"filters filters"}
+      display="flex"
+      flexDirection={{ base: "column" }}
+      justifyContent="start"
+      alignItems="start"
+      minH="100vh"
+      minW={["100vw"]}
+      gap={{ base: 10, md: 1 }}
+      fontWeight="bold"
+      bg="gray.100"
+    >
+      <GridItem
+        area={"filters"}
```

Comment:
General rule

if the values are static then please avoid wrapping them in culy braces
```suggestion
        area="filters"
```